warehouse/query-core/src/main/java/datawave/query/index/lookup/Union.java

```diff
@@ -43,110 +44,102 @@ public Union(Iterable<? extends IndexStream> children) {
-        boolean exceededValueThreshold = false;
+        
+        boolean delayedFromUnindexed = false; // did any delayed node come from an unindexed stream?
+        
         for (IndexStream stream : children) {
             
             if (log.isDebugEnabled()) {
                 childrenContextDebug.add(stream.getContextDebug());
             }
             
-            childrenCount++;
             if (log.isTraceEnabled()) {
-                log.trace("Union of " + stream.currentNode() + " " + stream.hasNext() + " " + JexlStringBuildingVisitor.buildQuery(stream.currentNode()));
-                log.trace("Union of " + stream + " " + stream.context());
-            }
-            
-            if (StreamContext.NO_OP == stream.context())
-                continue;
-            if (StreamContext.EXCEEDED_VALUE_THRESHOLD == stream.context()) {
-                exceededValueThreshold = true;
-            } else if (StreamContext.EXCEEDED_TERM_THRESHOLD == stream.context()) {
-                exceededTermThreshold = true;
-            } else if (StreamContext.UNINDEXED == stream.context()) {
-                this.childNodes.add(stream.currentNode());
-                this.delayedNodes.add(stream.currentNode());
-                unindexedField = true;
-                continue;
-            } else if (StreamContext.DELAYED_FIELD == stream.context()) {
-                this.childNodes.add(stream.currentNode());
-                this.delayedNodes.add(stream.currentNode());
-                delayedField = true;
-                continue;
+                log.trace("Union child " + JexlStringBuildingVisitor.buildQuery(stream.currentNode()) + " - " + stream.context() + " hasNext: "
+                                + stream.hasNext());
             }
             
             if (stream.hasNext()) {
-                this.children.add(stream);
-                this.childNodes.add(stream.currentNode());
-            } else {
                 switch (stream.context()) {
-                
-                    case INITIALIZED:
-                        // this should never be returned
-                        throw new RuntimeException("Invalid state in RangeStream");
-                    case IGNORED:
-                        childrenIgnored = true;
-                        this.childNodes.add(stream.currentNode());
-                        this.delayedNodes.add(stream.currentNode());
-                        break;
+                    case PRESENT:
+                    case VARIABLE:
                     case EXCEEDED_VALUE_THRESHOLD:
-                    case EXCEEDED_TERM_THRESHOLD:
-                        if (log.isTraceEnabled())
-                            log.trace("Adding current node to stream");
-                        /*
-                         * Helpful for debugging
-                         */
+                        // index streams with data are always added
+                        this.children.add(stream);
```

Comment:
is it intended to add this back into the same children set that is being looped over?

Reply:
It's not the same collection. The incoming collection of index stream is called `children`, which is different from the class-level priority queue called `children`. I'm renaming the provided collection of index streams to `streams`